cruise-control/src/test/java/com/linkedin/kafka/cruisecontrol/analyzer/RackAwareExcludedTopicsTest.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright 2017 LinkedIn Corp. Licensed under the BSD 2-Clause License (the "License"). See License in the project root for license information.
+ */
+
+package com.linkedin.kafka.cruisecontrol.analyzer;
+
+import com.linkedin.kafka.cruisecontrol.CruiseControlUnitTestUtils;
+import com.linkedin.kafka.cruisecontrol.analyzer.goals.Goal;
+import com.linkedin.kafka.cruisecontrol.analyzer.goals.RackAwareCapacityGoal;
+import com.linkedin.kafka.cruisecontrol.analyzer.goals.RackAwareGoal;
+import com.linkedin.kafka.cruisecontrol.config.KafkaCruiseControlConfig;
+import com.linkedin.kafka.cruisecontrol.common.ClusterProperty;
+import com.linkedin.kafka.cruisecontrol.common.RandomCluster;
+import com.linkedin.kafka.cruisecontrol.common.TestConstants;
+import com.linkedin.kafka.cruisecontrol.model.ClusterModel;
+
+import java.lang.reflect.Constructor;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+
+import java.util.Set;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.junit.runners.Parameterized.Parameters;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static org.junit.Assert.assertTrue;
+
+
+/**
+ * Unit test for testing rack awareness with excluded topics under fixed cluster properties.
+ */
+@RunWith(Parameterized.class)
+public class RackAwareExcludedTopicsTest {
```

Comment:
This test seems a little confusing. 

I think we should have some more straight forward and specific tests. For example:
1. Test a cluster model with 2 racks, a single topic with a single partition, but both replicas are in the same rack. Ensure when the topic is not excluded, the proposals contains the partition. When the topic is excluded, no proposal is generated.
2. Test the cases when an exception should be thrown and when an exception should not be thrown, depending on whether the violated topic is included or not.

Also, the test should be run on all the goals.